internal/workflow.go

```diff
@@ -308,10 +307,7 @@ func NewNamedSelector(ctx Context, name string) Selector {
 
 // NewWaitGroup creates a new WaitGroup instance.
 func NewWaitGroup(ctx Context) WaitGroup {
-	return &waitGroupImpl{
-		mu:       sync.Mutex{},
-		selector: NewSelector(ctx).(*selectorImpl),
-	}
+	return &waitGroupImpl{future: &futureImpl{channel: NewChannel(ctx).(*channelImpl)}}
```

Comment:
Why not NewFuture?

Reply:
I originally implemented using NewFuture(), but I did not want to deal with the returned Settable as well.  I can switch it back and ignore the Settable if it would be more readable.